singer/src/main/java/com/pinterest/singer/writer/KafkaWritingTask.java

```diff
@@ -116,7 +116,7 @@ public KafkaWritingTaskResult call() {
       OpenTsdbMetricConverter.incrGranular(SingerMetrics.BROKER_WRITE_FAILURE, 1, "broker=" + leaderNode);
       result = new KafkaWritingTaskResult(false, e);
     } finally {
-      if (!result.success) {
+      if (result != null && !result.success) {
```

Comment:
why do we need to add `result != null` check here?

Reply:
A Throwable happening before L93 will not be catched. `result` will not be initialized and causes result.success to throw an NPE that masks the actual Throwable from getting escalated.